lib/thrift/clients/binary.ex

```diff
@@ -0,0 +1,183 @@
+defmodule Thrift.Clients.Binary do
+  alias Thrift.Protocols.Binary
+  alias Thrift.TApplicationException
+
+  defmodule State do
+    defstruct host: nil, port: nil, tcp_opts: nil, timeout: 5000, sock: nil
+  end
+
+  use Connection
+
+  def init({host, port, tcp_opts, timeout}) do
+    s = %State{host: host,
+               port: port,
+               tcp_opts: tcp_opts,
+               timeout: timeout,
+               sock: nil}
+
+    {:connect, :init, s}
+  end
+
+  def start_link(host, port, tcp_opts, timeout \\ 5000) do
+    Connection.start_link(__MODULE__, {host, port, tcp_opts, timeout})
+  end
+
+  def close(conn), do: Connection.call(conn, :close)
+
+  def connect(_, %{sock: nil, host: host, port: port, tcp_opts: opts, timeout: timeout} = s) do
+    opts = Keyword.put_new(opts, :send_timeout, 1000)
+    case :gen_tcp.connect(host, port, [active: false, packet: 4, mode: :binary] ++ opts, timeout) do
+      {:ok, sock} ->
+        {:ok, %{s | sock: sock}}
+
+      {:error, _} ->
+        {:backoff, 1000, s}
```

Comment:
I was thinking of hard-coding a backoff strategy here using an increasing fibonacci sequence.